ext/bcmath/libbcmath/src/div.c

```diff
@@ -31,219 +31,428 @@
+	for (size_t i = 0; i < quot_arr_size - 1; i++) {
+		if (n1_vector[n1_top_index - i] < n2_vector) {
+			quot_vector[quot_top_index - i] = 0;
+			/* n1_vector[n1_top_index - i] is always a smaller number than n2_vector. Therefore, there will be no overflow. */
+			n1_vector[n1_top_index - i - 1] += n1_vector[n1_top_index - i] * BC_VECTOR_BOUNDARY_NUM;
+			continue;
+		}
+		quot_vector[quot_top_index - i] = n1_vector[n1_top_index - i] / n2_vector;
+		n1_vector[n1_top_index - i] -= n2_vector * quot_vector[quot_top_index - i];
+		n1_vector[n1_top_index - i - 1] += n1_vector[n1_top_index - i] * BC_VECTOR_BOUNDARY_NUM;
+		n1_vector[n1_top_index - i] = 0;
+	}
+	/* last */
+	quot_vector[0] = n1_vector[0] / n2_vector;
+}
 
-	if (digit == 0) {
-		memset(result, 0, size);
-	} else {
-		if (digit == 1) {
-			memcpy(result, num, size);
-		} else {
-			/* Initialize */
-			nptr = (unsigned char *) (num + size - 1);
-			rptr = (unsigned char *) (result + size - 1);
-			carry = 0;
-
-			while (size-- > 0) {
-				value = *nptr-- * digit + carry;
-				*rptr-- = value % BASE;
-				carry = value / BASE;
-			}
+/*
+ * Used when the number of elements in the n2 vector is 2 or more.
+ */
+static inline void bc_standard_div(
+	BC_VECTOR *n1_vector, size_t n1_arr_size,
+	BC_VECTOR *n2_vector, size_t n2_arr_size, size_t n2_len,
+	BC_VECTOR *quot_vector, size_t quot_arr_size
+) {
+	size_t i, j;
+	size_t n1_top_index = n1_arr_size - 1;
+	size_t n2_top_index = n2_arr_size - 1;
+	size_t quot_top_index = quot_arr_size - 1;
+
+	BC_VECTOR div_carry = 0;
+
+	/*
+	 * Consider the error E between the actual quotient and the temporary quotient.
+	 * For example, the quotient of 240 / 121 is 1, but if calculate the quotient only using the high-order digits (24 / 12),
+	 * it will be 2. In this case, 2 is the temporary quotient. Here, the error E is 1.
+	 * Also note that for example 2400000 / 120, there will be 5 divisions.
+	 *
+	 * Another example: Calculating 99900000000 / 10009999 with radix 10000.
+	 * The actual quotient is 9980, but if calculate it using only the high-order digits:
+	 * 999 / 1000 = 0, Numbers that are not divisible are carried forward to the next division.
+	 * 9990000 / 1000 = 9990 (Here, the temporary quotient is 9990 and the error E is 10.)
+	 *
+	 * If calculate the temporary quotient using only one array of n1 and n2, the error E can be larger than 1.
+	 * In other words, in the restoring division, the count of additions for restore increases significantly.
+	 *
+	 * Therefore, in order to keep the error within 1 and to limit the number of additions required for restoration to one
+	 * at most, adjust the number of high-order digits used to calculate the temporary quotient as follows.
+	 * - Adjust the number of digits of n2 used in the calculation to BC_VECTOR_SIZE + 1 digit. The missing digits are
+	 *   filled in from the next array element.
+	 * - Add digits to n1 in the same way as the number of digits adjusted by n2.
```

Comment:
So, doing the same for `n1` as for `n2`?

Reply:
Adjust `n1` the same number of digit adjustments made to `n2`. The important thing here is that instead of increasing the number of digits in `n1` to `BC_VECTOR_SIZE + 1` digit, just increase `n1` by the same amount as `n2` increases.